src/test/java/com/google/devtools/build/lib/remote/RemoteSpawnRunnerTest.java

```diff
@@ -252,47 +252,17 @@ public void cachableSpawnsShouldBeCached_localFallback() throws Exception {
     verify(cache).upload(any(), any(), any(), any(), any(), any());
   }
 
-  @Test
-  public void noRemoteExecExecutesLocallyButUsesCache() throws Exception {
-    // Test that if the NO_REMOTE_EXEC execution requirement is specified, the spawn is executed
-    // locally, but its result is still uploaded to the remote cache.
-
-    remoteOptions.remoteAcceptCached = true;
-    remoteOptions.remoteUploadLocalResults = true;
-
-    RemoteSpawnRunner runner = spy(newSpawnRunner());
-
-    SpawnResult res =
-        new SpawnResult.Builder()
-            .setStatus(Status.SUCCESS)
-            .setExitCode(0)
-            .setRunnerName("test")
-            .build();
-    when(localRunner.exec(any(Spawn.class), any(SpawnExecutionContext.class))).thenReturn(res);
-
-    Spawn spawn =
-        simpleSpawnWithExecutionInfo(ImmutableMap.of(ExecutionRequirements.NO_REMOTE_EXEC, ""));
-    SpawnExecutionContext policy =
-        new FakeSpawnExecutionContext(spawn, fakeFileCache, execRoot, outErr);
-
-    SpawnResult result = runner.exec(spawn, policy);
-
-    assertThat(result.exitCode()).isEqualTo(0);
-    assertThat(result.status()).isEqualTo(Status.SUCCESS);
-    verify(localRunner).exec(eq(spawn), eq(policy));
-    verify(runner)
-        .execLocallyAndUpload(
-            eq(spawn), eq(policy), any(), any(), any(), any(), /* uploadLocalResults= */ eq(true));
-    verify(cache).upload(any(), any(), any(), any(), any(), any());
-  }
-
   @Test
   public void failedLocalActionShouldNotBeUploaded() throws Exception {
     // Test that the outputs of a locally executed action that failed are not uploaded.
 
+    remoteOptions.remoteLocalFallback = true;
     remoteOptions.remoteUploadLocalResults = true;
 
-    RemoteSpawnRunner runner = spy(newSpawnRunnerWithoutExecutor());
+    RemoteSpawnRunner runner = spy(newSpawnRunner());
+
+    // Throw an IOException to trigger the local fallback.
+    when(executor.executeRemotely(any(ExecuteRequest.class))).thenThrow(IOException.class);
```

Comment:
The name and description of the test don't fit its implementation. What is this testing?